mobile/App.js

```diff
@@ -6,25 +6,111 @@
  * @flow strict-local
  */
 
-import React, {useState} from 'react';
+import React, {useState, useRef, useEffect} from 'react';
 import {SafeAreaView} from 'react-native-safe-area-context';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
 import HomeView from './src/views/HomeView';
-import ScanView from './src/views/ScanView';
 import VendorView from './src/views/VendorView';
 import RegistrationView from './src/views/RegistrationView';
 import {NavigationContainer} from '@react-navigation/native';
 import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
 import ScreenOption from './src/utilites/ScreenOption';
-import {globalStyleSheet} from './src/utilites/Theme';
 import MyICardView from './src/views/MyICardView';
-
+import { AppState } from "react-native";
 import AuthContext from './src/context/AuthContext';
+import { storeDate } from './src/utilites/StoreDate';
+import { storeUser } from './src/utilites/StoreUser';
 
 const Tab = createBottomTabNavigator();
 
 const App = () => {
   const [user, setUser] = useState(null);
+  const [check, setCheck] = useState(false);
   const value = {user, setUser};
+  const appState = useRef(AppState.currentState);
+  
+  const logoutCheck = async(userData = null,startup = false) => {
+    
+    try{
+        const lastDate = parseInt(await AsyncStorage.getItem('lastOpened'));    
+        if(lastDate != null){
+            //change num in if statement below to a month in milliseconds
+            if(new Date().getTime() - lastDate > 2629800000){
+                setUser(null)
+                storeUser(null)
+            }
+            else{
+                if(startup){
+                  setUser(userData)
+                }
+                await storeDate()
+            }
+        }
+    } catch(error){
+        logoutCheck()
+    }
+}
+  useEffect(() => {
+    const checkUser = async() => {
+      const userData = JSON.parse(await AsyncStorage.getItem('userData'))
+      if(userData != null){
+        logoutCheck(userData, true)
+      }
+
+    } 
+    checkUser()
+  }, [])
+  
+  
+  useEffect(() => {
+    const subscription = AppState.addEventListener("change", nextAppState => {
+      
+      if (
+        appState.current.match(/active/) &&
+        nextAppState === "background"
+      ) {
+        
+        if(user != null){ 
+          storeDate()
+        }
+      }
+
+      appState.current = nextAppState;
+      
+    });
+
+    return () => {
+      subscription.remove();
+    };
+  }, [user]);
+  useEffect(() => {
+    const subscription = AppState.addEventListener("change", nextAppState => {    
+      appState.current = nextAppState;
+      if(appState.current === 'active'){
+        
+        if(user != null){         
```

Comment:
You can combine this into one if statement that just calls logoutCheck. Then we can also remove the other useEffect and useState that use the check.

Reply:
so keep both if conditions but combine them into one line? I copied the base appstate code from the documentation. For some reason (Sometimes) subtle changes to the appstate code messes up the functionality for no reason. That's why I kept stuff as it is in the react native documentation. Playing it safe technically

Reply:
Yea its just a logic change so there is no need to worry about breakage. (Can verify yourself if you have doubts). Double if statement without an else is unnecessary.